vizro-ai/src/vizro_ai/task_pipeline/pipeline.py

```diff
@@ -0,0 +1,60 @@
+"""Task Pipeline."""
+
+from typing import Any, Dict, List, Union
+
+from vizro_ai.chains._llm_models import LLM_MODELS
+
+
+class Pipeline:
+    """A pipeline to manage the flow of tasks in a sequence."""
+
+    def __init__(self, llm: LLM_MODELS):
+        """Initialize the Pipeline.
+
+        Args:
+            llm: The LLM instance to be used by components in the pipeline.
+        """
+        self.llm = llm
+        self.components = []
+        self.components_instances = {}
+
+    def add(self, component_class, input_keys: Union[List, None] = None, output_key: Union[str, None] = None):
```

Comment:
```suggestion
    def add(self, component_class, input_keys: Optional[List[str]] = None, output_key: Optional[str] = None):
```
https://stackoverflow.com/questions/51710037/how-should-i-use-the-optional-type-hint